### ISSUE TITLE
Tooltips aren't accessible

### ISSUE THREAD

**Issue by jamesjacobs** [jadu/pulsar#1019]
**Describe the bug**
Existing tooltips don't offer the same experience to SR users as the tooltip contents aren't read when the triggering element is focussed/hovered. The only way for a SR user to find/read the tooltip contents is to focus the trigger and use the SR to navigate text on the page. Additionally they do not currently meet WCAG 1.4.13 at AA.

**Ideal behaviour**
There's a number of things to consider here:

1. The tooltip contents should be read when the triggering element is focussed. Depending on the use case this could be with either `aria-labbelledby` (in the case of a button with a count bubble) or `aria-describedby` for general tooltip usage.

2. To meet 1.4.13 (Content on Hover or Focus AA), the tooltip must be:
 **Dismissable** -  without moving the pointer or KB focus. So we need to be able to close the tooltip with ESC.
**Hoverable**  - the tooltip contents need to stay visible whilst the are hovered over 
**Persistent** - the tooltip content remains visible until the hover or focus trigger is removed, the user dismisses it or the information is no longer valid.

3. As popovers are powered by the existing tooltip.js, for backwards compatibility it's worth adding new tooltip functionality to a new JS module with a different `data` attribute hook.

**Proposed solution**
Example outputted markup:
```
<button aria-describedby="guid-1234">  
  Example button with tooltip
</button>  
<div role="tooltip" id="guid-1234">Tooltip contents</div>  
```

The `aria-describedby` and `role="tooltip"` `div` could be added via JS using a data attribute for both the selector and contents. For example:

```
<button data-tooltip="Tooltip contents">  
  Example button with tooltip
</button> 
```

We'd need to ensure there's no clash between existing `tooltip.js` functionality and new functionality. Potentially by removing the `tooltip.js` markup / unbinding any events/listeners in case both new and old attributes are used. We could also remove any `title` attribute on the triggering element too as they don't offer much in the way of accessibility.

It may be necessary to wrap the triggering element and the added tooltip `div` for positioning reasons.

**Comment by jackbentley** [comment]
In addition, I believe tooltips don't work well on mobile devices or small screens.

On small screens with the position set to right, on a full-width element, the tooltip is off-screen.
